en/docs/security-guidelines/secure-engineering-guidelines/dynamic-analysis-with-owasp-zap.md

```diff
@@ -16,199 +16,199 @@ This document provides details of all necessary steps for configuring the OWASP
+![Placeholder](../../assets/images/secure-coding-guidelines/zap-05.png)
 
 If any exception for **localhost, 127.0.0.1** or the hostname of the WSO2 server you are trying to scan is given in **No Proxy for:** text box, remove them so ZAP can detect the traffic flow of that as well. 
 
-Now, go to firefox and access the WSO2 server. You will see the SSL warning below because the traffic goes through ZAP proxy. Add an exception to the site in the browser.
+Now, go to firefox and access the WSO2 server. You will see the SSL warning below because the traffic goes through the ZAP proxy. Add an exception to the site in the browser.
 
-![Placeholder]({{#base_path#}}/assets/images/secure-coding-guidelines/zap-06.png){ .post-image }
+![Placeholder](../../assets/images/secure-coding-guidelines/zap-06.png)
 
-Now the WSO2 Server URL should be opened in the browser. Also it should be listed in ZAP under the sites. 
+Now the WSO2 Server URL should be opened in the browser. Also, it should be listed in ZAP under the sites. 
 
-![Placeholder]({{#base_path#}}/assets/images/secure-coding-guidelines/zap-07.png){ .post-image }
+![Placeholder](../../assets/images/secure-coding-guidelines/zap-07.png)
 
-Select the Mode of the scan as **Protected Mode**. With this, you can choose which sites to be used for attacking. *(For example, if you are trying out the Federated Authentication scenario with WSO2 Identity Server and Facebook, under the Sites in ZAP tool, the facebook website also will be listed to be attacked. In the scope of scanning, external websites should be removed. With the Protected Mode, you have the capability to select the Sites that should be attacked by ZAP tool)*
+Select the Mode of the scan as **Protected Mode**. With this, you can choose which sites to be used for attacking. *(For example, if you are trying out the Federated Authentication scenario with WSO2 Identity Server and Facebook, under the Sites in ZAP tool, the Facebook website also will be listed to be attacked. In the scope of scanning, external websites should be removed. With the Protected Mode, you have the capability to select the Sites that should be attacked by the ZAP tool)*
 
 
 ## Scanning Process
 
 ### Excluding the Server Logout from Spider
-When we run the ZAP scan with an active user session, if ZAP executes the action to logout from the server in the middle of the scan, the actions that should be performed with a logged in user session would not be performed after that (because active session is removed with logout action ). In order to avoid that, we need to exclude the logout action from the Spider.
+When we run the ZAP scan with an active user session, if ZAP executes the action to logout from the server in the middle of the scan, the actions that should be performed with a logged in user session would not be performed after that (because the active session is removed with logout action ). In order to avoid that, we need to exclude the logout action from the Spider.
```

Comment:
```suggestion
When we run the ZAP scan with an active user session, if ZAP executes the action to logout from the server in the middle of the scan, the actions that should be performed with a logged in user session would not be performed after that (because the active session is removed with the logout action ). In order to avoid that, we need to exclude the logout action from the Spider.
```